SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -735,7 +735,11 @@ bool RunSchemaCompiler()
 	FString PluginDir = FSpatialGDKServicesModule::GetSpatialGDKPluginDirectory();
 
 	// Get the schema_compiler path and arguments
-	FString SchemaCompilerExe = FPaths::Combine(PluginDir, TEXT("SpatialGDK/Binaries/ThirdParty/Improbable/Programs/schema_compiler.exe"));
+#if PLATFORM_WINDOWS
+    FString SchemaCompilerExe = FPaths::Combine(PluginDir, TEXT("SpatialGDK/Binaries/ThirdParty/Improbable/Programs/schema_compiler.exe"));
```

Comment:
Might be overkill but changing this to be a function that uses the duplicated `FString SchemaCompilerExe = FPaths::Combine(PluginDir, TEXT("SpatialGDK/Binaries/ThirdParty/Improbable/Programs/schema_compiler"))` part and then appends a `PlatformExecutableExtension` (using `FPaths::SetExtension`) if we're in `PLATFORM_WINDOWS` might be cleaner. 
Actually the second part can be done without a function :D

Reply:
my main concern with that is that people will still be able to call the field directly, which might cause issues if they only test one platform ;)  this way they can only call the one that is already correct for the platform. Though I guess I could move it into a separate file with the fields being private and only accessible via a getter function? and simply move all of our exe fields into that file